parameters.go

```diff
@@ -64,7 +65,7 @@ func (p *Processor) NewParameters(ctx context.Context, input *image.ImageFile, q
 	}
 
 	if format != input.Format() {
-		index := len(filepath) - len(input.Format())
+		index := len(filepath) - len(input.Format()) - 1
```

Comment:
@Thommas2316 not sure about this one, why the `- 1`?

If `foo.jpg` is the filepath then it will be `foo.` and concat the new format.